src/java/org/jivesoftware/openfire/plugin/util/cluster/ClusterPacketRouter.java

```diff
@@ -37,13 +38,16 @@
 
     private static Logger logger = LoggerFactory.getLogger(ClusterPacketRouter.class);
 
-    public boolean routePacket(byte[] nodeID, JID receipient, Packet packet) {
+    public boolean routePacket(byte[] nodeID, JID recipient, Packet packet) {
         // Send the packet to the specified node and let the remote node deliver the packet to the recipient
         try {
-            CacheFactory.doClusterTask(new RemotePacketExecution(receipient, packet), nodeID);
+            if (!ClusterManager.isClusterMember(nodeID)) {
+                return false;
+            }
+            CacheFactory.doClusterTask(new RemotePacketExecution(recipient, packet), nodeID);
             return true;
-        } catch (IllegalStateException  e) {
-            logger.warn("Error while routing packet to remote node: " + e);
+        } catch (Exception  e) {
+            logger.warn("Error while routing packet to remote node",e);
```

Comment:
How exactly do we reach a state where things are routed to non-existing nodes? If this hints at a bug in the Openfire implementation, then I think it warrants logging to 'error' rather than 'warn'.

Also, this is more nitpicking, but it would be useful to log the identifier of the non-existing node that this is being routed to.

```suggestion
            logger.error("Error while routing packet to remote node: {}", nodeID == null ? "(null)" : new String(nodeID, java.nio.charset.StandardCharsets.UTF_8), e);
```

Reply:
Hi @guusdk,

the node id is already part of the exception message:

`
2020.11.03 17:07:45.965 ERROR [hz.openfire.cached.thread-4] (org.jivesoftware.openfire.spi.RoutingTableImpl:279) - Primary packet routing failed
java.lang.IllegalArgumentException: Requested node 5ed5407e-e7fb-4b87-8ef0-9aff16dc07a0 not found in cluster
	at org.jivesoftware.openfire.plugin.util.cache.ClusteredCacheFactory.doClusterTask(ClusteredCacheFactory.java:397) ~[hazelcast-2.5.1-SNAPSHOT.jar!/:?]
	at org.jivesoftware.util.cache.CacheFactory.doClusterTask(CacheFactory.java:701) ~[xmppserver-4.6.0-SNAPSHOT.jar:4.6.0-SNAPSHOT]
`

Reply:
But I see it would at least be logged on all other exceptions as well.